addons/mail/static/src/core/common/message.xml

```diff
@@ -41,7 +41,7 @@
                             </small>
                             <small t-if="isPersistentMessageFromAnotherThread" t-on-click.prevent="openRecord" class="ms-1 text-500">
                                 <t t-if="message.thread.model !== 'discuss.channel'">
-                                    on <a t-att-href="message.resUrl"><t t-esc="message.thread.displayName"/></a>
+                                    on <a t-if="message.thread.displayName" t-att-href="message.resUrl" t-esc="message.thread.displayName"/><em class="pe-1 text-decoration-line-through" t-else="">Deleted document</em>
```

Comment:
If there is no thread `isPersistentMessageFromAnotherThread` is false, and we should never reach this line?

Reply:
It reaches this part: opening in mailbox, message is not transient, and there's `message.thread` (because `message` still provide `res_id` and `model`, thus this inserts a `thread`).

Reply:
We should be cleaning this too, `res_id` and `model` shouldn't be returned when the corresponding record no longer exists.